src/IceRpc/Transports/Slic/Internal/SlicConnection.cs

```diff
@@ -458,12 +459,11 @@ public async ValueTask<IMultiplexedStream> CreateStreamAsync(
 
     public ValueTask DisposeAsync()
     {
-        TryClose(new IceRpcException(IceRpcError.OperationAborted), "The connection was disposed.");
-
         lock (_mutex)
         {
             _disposeTask ??= PerformDisposeAsync();
         }
+        TryClose(new IceRpcException(IceRpcError.OperationAborted), "The connection was disposed.");
```

Comment:
I don't understand this update.

For me, it makes more sense to close and then dispose - not the other way around.

It also implies that _dispatchTask is not null _only_ once _isClosed is true.

Reply:
Because TryClose calls `_acceptStreamChannel.Writer.TryComplete(exception)` which will complete the pending accept tasks before dispose.

Reply:
I don't see the issue. This seems fine.

Reply:
The issue is that the pending accept task will fail with OperationAborted and we want to align it with QUIC which fails with ObjectDisposedException.

Reply:
Ok, what about moving the TryClose call to just after "await Task.Yield" in PerformDisposeAsync?

I think it would be more correct. _disposeTask is set but we haven't done anything else yet. And your new code is checking `_disposeTask is not null`.